src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -222,30 +222,38 @@ def _validate_vm_create_availability_set(namespace):
             name=name)
 
 
-def _validate_vm_create_vnet(namespace):
+def _validate_vm_create_vnet(namespace, for_scale_set=False):
 
     vnet = namespace.vnet_name
     subnet = namespace.subnet
     rg = namespace.resource_group_name
     location = namespace.location
     nics = getattr(namespace, 'nics', None)
 
-    if not vnet and not subnet and not nics:
+    if not vnet and not subnet and not nics:  # pylint: disable=too-many-nested-blocks
         # if nothing specified, try to find an existing vnet and subnet in the target resource group
         from azure.mgmt.network import NetworkManagementClient
         from azure.cli.core.commands.client_factory import get_mgmt_service_client
         client = get_mgmt_service_client(NetworkManagementClient).virtual_networks
 
-        # find VNET in target resource group that matches the VM's location and has a subnet
+        # find VNET in target resource group that matches the VM's location with a matching subnet
         for vnet_match in (v for v in client.list(rg) if v.location == location and v.subnets):
 
             # 1 - find a suitable existing vnet/subnet
-            subnet_match = next(
-                (s.name for s in vnet_match.subnets if s.name.lower() != 'gatewaysubnet'), None
-            )
-            if not subnet_match:
+            subnet = None
+            subnet_matches = [s for s in vnet_match.subnets if s.name.lower() != 'gatewaysubnet']
+            if subnet_matches:
```

Comment:
@tjprescott  I tried to use only one list comprehension so to simplify the code. But I am fine to switch to your logic. Could you take another look? Note, we don't need `or []` because the outer loop has already filtered away empty or none cases